### ISSUE TITLE
Navigation fragment freezing

### ISSUE THREAD

**Issue by rav005** [mapbox/mapbox-navigation-android#3589]
### Version
Android: 10
mapbox-android-sdk:9.4.0
mapbox-android-navigation: 0.42.6
mapbox-android-navigation-ui:0.42.6

### Steps to trigger behavior
Activity starts with SupportMapFragment and custom fragment. A event from custom fragment replaces SupportMapFragment  with NavigationView fragment and starts navigation to destination points. Which works fine but after 5-10 minutes the navigation map freezes but hear navigation instructions and it updates instructions header. I have to go to home screen and resume activity to resume navigation map update.

Logs:
```
I/System.out: (HTTPLog)-Static: isSBSettingEnabled false
I/System.out: (HTTPLog)-Static: isSBSetting
![navigation issue](https://user-images.githubusercontent.com/13260867/94052270-b3227600-fda6-11ea-813d-1a33b01bcbe3.jpg)
Enabled false
I/reon.cit.drive: Background concurrent copying GC freed 46486(3456KB) AllocSpace objects, 21(1200KB) LOS objects, 49% free, 7565KB/14MB, paused 1.217ms total 120.510ms
I/Choreographer: Skipped 3709 frames!  The application may be doing too much work on its main thread.
D/ViewRootImpl@2270598[TransitManifestView]: Relayout returned: old=(0,0,1200,1920) new=(0,0,1200,1920) req=(1200,1920)8 dur=16 res=0x1 s={false 0} ch=false
D/ViewRootImpl@2270598[TransitManifestView]: stopped(false) old=true
D/SurfaceView: windowStopped(false) false com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906} of ViewRootImpl@2270598[TransitManifestView]
D/ViewRootImpl@2270598[TransitManifestView]: stopped(false) old=false
D/ViewRootImpl@2270598[TransitManifestView]: stopped(true) old=false
D/SurfaceView: windowStopped(true) false com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906} of ViewRootImpl@2270598[TransitManifestView]
E/Mbgl-NativeMapView: You're calling `getCameraValues` after the `MapView` was destroyed, were you invoking it after `onDestroy()`?
D/ViewRootImpl@2270598[TransitManifestView]: stopped(false) old=true
D/SurfaceView: windowStopped(false) false com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906} of ViewRootImpl@2270598[TransitManifestView]
D/ViewRootImpl@2270598[TransitManifestView]: stopped(false) old=false
I/Choreographer: Skipped 165 frames!  The application may be doing too much work on its main thread.
D/SurfaceView: onWindowVisibilityChanged(0) true com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906} of ViewRootImpl@2270598[TransitManifestView]
D/ViewRootImpl@2270598[TransitManifestView]: Relayout returned: old=(0,0,1200,1920) new=(0,0,1200,1920) req=(1200,1920)0 dur=16 res=0x7 s={true 3171760128} ch=true
D/OpenGLRenderer: createReliableSurface : 0xef76ef40(0xbd0d3800)
D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
D/SurfaceView: surfaceCreated 1 #5 com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906}
D/SurfaceView: surfaceChanged (1200,906) 1 #5 com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.MapboxGLSurfaceView{159c8ac V.E...... ........ 0,0-1200,906}
D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
W/libEGL: EGLNativeWindowType 0xbd6de008 disconnect failed
D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
D/Mbgl-HttpRequest: [HTTP] Request with response = 304: Not Modified
D/Mbgl-HttpRequest: [HTTP] Request with response = 304: Not Modified
V/Mbgl-HttpRequest: [HTTP] Request was successful (code = 200).
D/Mbgl-HttpRequest: [HTTP] Request with response = 304: Not Modified
D/Mbgl-HttpRequest: [HTTP] This request was cancelled (https://api.mapbox.com/v4/mapbox.mapbox-incidents-v1/12/1139/1493.vector.pbf?access_token=pk.eyJ1IjoibmFtc3lzaW5jIiwiYSI6ImNrZXgyb3k4eDJvdWkycm4xODljNWhldnkifQ.tc1CwoaogqbgJkI1XJMKug&sku=100kffodptoe911eb5c448345baaf4ebfc23e431b12). This is expected for tiles that were being prefetched but are no longer needed for the map to render.
D/Mbgl-HttpRequest: [HTTP] Request with response = 304: Not Modified
D/Mbgl-HttpRequest: [HTTP] This request was cancelled (https://api.mapbox.com/v4/mapbox.mapbox-incidents-v1/12/1139/1492.vector.pbf?access_token=pk.eyJ1IjoibmFtc3lzaW5jIiwiYSI6ImNrZXgyb3k4eDJvdWkycm4xODljNWhldnkifQ.tc1CwoaogqbgJkI1XJMKug&sku=100kffodptoe911eb5c448345baaf4ebfc23e431b12). This is expected for tiles that were being prefetched but are no longer needed for the map to render.
D/Mbgl-HttpRequest: [HTTP] This request was cancelled (https://api.mapbox.com/v4/mapbox.mapbox-incidents-v1/8/71/93.vector.pbf?access_token=pk.eyJ1IjoibmFtc3lzaW5jIiwiYSI6ImNrZXgyb3k4eDJvdWkycm4xODljNWhldnkifQ.tc1CwoaogqbgJkI1XJMKug&sku=100kffodptoe911eb5c448345baaf4ebfc23e431b12). This is expected for tiles that were being prefetched but are no longer needed for the map to render.
    [HTTP] This request was cancelled (https://api.mapbox.com/v4/mapbox.mapbox-incidents-v1/12/1139/1494.vector.pbf?access_token=pk.eyJ1IjoibmFtc3lzaW5jIiwiYSI6ImNrZXgyb3k4eDJvdWkycm4xODljNWhldnkifQ.tc1CwoaogqbgJkI1XJMKug&sku=100kffodptoe911eb5c448345baaf4ebfc23e431b12). This is expected for tiles that were being prefetched but are no longer needed for the map to render.
D/Mbgl-HttpRequest: [HTTP] This request was cancelled (https://api.mapbox.com/v4/mapbox.mapbox-incidents-v1/12/1141/1492.vector.pbf?access_token=pk.eyJ1IjoibmFtc3lzaW5jIiwiYSI6ImNrZXgyb3k4eDJvdWkycm4xODljNWhldnkifQ.tc1CwoaogqbgJkI1XJMKug&sku=100kffodptoe911eb5c448345baaf4ebfc23e431b12). This is expected for tiles that were being prefetched but are no longer needed for the map to render.
```
![navigation issue](https://user-images.githubusercontent.com/13260867/94052313-c2092880-fda6-11ea-82bd-ec6d29d9758b.jpg)

**Comment by rav005** [comment]
I created sample application with code provided on mapbox tutorials (https://docs.mapbox.com/help/tutorials/android-navigation-sdk/) and still having issues. It looks like the issue only occurs in tablet device (using Samsung Galaxy Tab A), works fine on phones.

**Comment by zenyagami** [comment]
I can confirm the same issue , sometimes when i move the map when is under the navigation,  using the examples (ComponentNavigation) the navigation lags and takes a lot of time

**Comment by Guardiola31337** [comment]
Hey @rav005 and @zenyagami thanks for reaching out and report your issue.

Would you mind re-testing / confirming if OP is still an issue in 1.x (latest stable available is [`1.3.0`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v1.3.0))? You can review the developer [documentation](https://docs.mapbox.com/android/beta/navigation/overview/) to start building with the Mapbox Navigation SDK v1.0 for Android and check out the [migration guide](https://github.com/mapbox/mapbox-navigation-android/wiki/1.0-Navigation-SDK-Migration-Guide) to transition your project from the "legacy" Navigation SDK to the 1.0 version and let us know if you have any questions or run into issues. We would love your feedback.

**Comment by rav005** [comment]
@Guardiola31337 The new navigation SDK works and found out why the old one wasn't working as well. The issue was the device I was using didn't have Gyroscope. Your navigation SDK should clearly log these errors.
